19.09 -> 20.03

rissson opened the pull request:
This pull request updates shabka to the new release of NixOS. For now, both 19.09 and 20.03 are available, but 19.09 will be removed and default release will be set to 20.03.

This adds/updates/deletes (non-exhaustive list):

* [UPDATE] nixpkgs unstable
* [ADD] nixpkgs 20.03
* [UPDATE] home-manager unstable
* [ADD] home-maanger 20.03
* [DELETE] nixpkgs 19.09
* [DELETE] home-manager 19.09

For reference, the merge request for my dotshabka can be found [here](https://gitlab.com/lama-corp/infra/dotshabka/-/merge_requests/29)

kalbasit commented on modules/nixos/general/keyboard.nix:
```diff
@@ -49,11 +49,11 @@ with lib;
-                         then layout.layout
-                         else (if layout.variant == "bepo" then "fr-bepo" else layout.variant);
+    console.earlySetup = cfg.enableAtBoot;
+    console.keyMap = let layout = layouts."${builtins.head cfg.layouts}"; in
```
Are these working on 19.03 or are these only 20.03?

rissson replied:
They are only 20.03

rissson replied:
Is it a blocker?

kalbasit replied:
It is for keeping 19.03. So you need to either split on the version or just remove 19.03 and make 20.03 default. I don't mind the latter.

rissson replied:
I will remove 19.09

rissson commented:
`programs.ssh.askPassword` conflicts between the one defined by `gnome3` and the one defined by `plasma5`. We probably want to choose a default and `mkForce` set it in shabka.

kalbasit commented:
I don't use either. Might be a good opportunity to clean up things we do not currently use.

rissson commented:
> I don't use either. Might be a good opportunity to clean up things we do not currently use.

I created #326 for that purpose. I think we don't need to rework how `shabka.windowManager` works, especially if we plan to start clean.

rissson commented:
Virtualbox does not build on 20.03. Adding patches for https://github.com/NixOS/nixpkgs/pull/82693

I will also open PRs for flashplayer tomorrow

rissson commented:
> I will also open PRs for flashplayer tomorrow

My bad, everything is up-to-date in nixpkgs. nixpkgs-channels isn't up-to-date probably

rissson commented:
All my hosts build and have been pushed to cachix. Switching config on hedgehog now

rissson commented:
```
updating GRUB 2 menu...
installing the GRUB 2 EFI boot loader into /boot/efi...
Installing for x86_64-efi platform.
You have a memory leak (not released memory pool):
 [0x201ce80] dtree
 [0x201ead0] dtree
Internal error: Unreleased memory pool(s) found.
You have a memory leak (not released memory pool):
 [0x201ce80] dtree
 [0x201ead0] dtree
Internal error: Unreleased memory pool(s) found.
You have a memory leak (not released memory pool):
 [0x201ce80] dtree
 [0x201ead0] dtree
Internal error: Unreleased memory pool(s) found.
You have a memory leak (not released memory pool):
 [0x201ce80] dtree
 [0x201ead0] dtree
Internal error: Unreleased memory pool(s) found.
You have a memory leak (not released memory pool):
 [0x201ce80] dtree
 [0x201ead0] dtree
Internal error: Unreleased memory pool(s) found.
You have a memory leak (not released memory pool):
 [0x201ce80] dtree
 [0x201ead0] dtree
Internal error: Unreleased memory pool(s) found.
Could not prepare Boot variable: No space left on device
/nix/store/dfg84530sl471r3qaxh1qd1byd22yljp-grub-2.04/sbin/grub-install: error: efibootmgr failed to register the boot entry: Input/output error.
/nix/store/7hr7fws5y4a8md49syz4rz7j229q8pk3-install-grub.pl: installation of GRUB EFI into /boot/efi failed
warning: error(s) occurred while switching to the new configuration
```
But my `/boot/efi` has 549M free

rissson commented:
Okay so I got my 20.03 booted. Report so far:
* my graphics driver seems to not have loaded :/
* the fonts of terminals and polybar and rofi and i3 are broken
* I can't resize the font in urxvt
* I had to hack around to get lightdm to let me login. Apparently the way {display,desktop,window} managers work has been completely reworked in 20.03. I will work on that tomorrow and make i3 the only option in shabka (for now) to get this fixed.

rissson commented:
>     * my graphics driver seems to not have loaded :/

Fixed by adding `intel` to https://nixos.org/nixos/manual/options.html#opt-services.xserver.videoDrivers

rissson commented:
> the fonts of terminals and polybar and rofi and i3 are broken

This has been fixed

> I had to hack around to get lightdm to let me login. Apparently the way {display,desktop,window} managers work has been completely reworked in 20.03. I will work on that tomorrow and make i3 the only option in shabka (for now) to get this fixed.

This too

rissson commented:
I currently don't have any other problems with this release on `hedgehog`. Will switch my NAS to it sometime soon

kalbasit commented:
Alright, I'll test my systems now with it! Thank you, @rissson

rissson commented:
My nas has been switched without any problems

kalbasit commented:
My `home-manager-yl` is failing with this error:

```
error: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name ca.desrt.dconf was not provided by any .service files
```

rissson commented:
> My `home-manager-yl` is failing with this error:
> 
> ```
> error: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name ca.desrt.dconf was not provided by any .service files
> ```

I have something like this in my system configuration:

```
services.dbus.packages = with pkgs; [ gnome3.dconf ];
```

kalbasit commented:
@rissson I added a new NixOS module to do just that for us, and I added an assertion on home's gtk option. Everything seems to work fine on my NixOS workstation. I'll do Darwin now to make sure it's still good.

kalbasit commented:
Oh, I can't seem to login by typing a BS password and then tap my Yubikey anymore. Not a blocker for the merge though I can fix it later.

rissson commented:
> Oh, I can't seem to login by typing a BS password and then tap my Yubikey anymore. Not a blocker for the merge though I can fix it later.

I had that problem too. It seems either lightdm or pam won't open a session when entering a password. However, I was able to login by turning on autologin

rissson commented:
> @rissson I added a new NixOS module to do just that for us, and I added an assertion on home's gtk option. Everything seems to work fine on my NixOS workstation. I'll do Darwin now to make sure it's still good.

Followed those changes in my dotshabka.

kalbasit commented:
> > Oh, I can't seem to login by typing a BS password and then tap my Yubikey anymore. Not a blocker for the merge though I can fix it later.
> 
> I had that problem too. It seems either lightdm or pam won't open a session when entering a password. However, I was able to login by turning on autologin

I have it enabled as well, but since I lock my workstation anytime I walk away I now have to type in my password when I get back. Annoying but not a blocker, we can figure it out as we go.

kalbasit commented:
Darwin seems to work fine as well. Pushing cache and will restart builds.

rissson commented:
I will switch my last host tonight at 00:00 CET, so I don't get downtime in the middle of the day
